### ISSUE TITLE
Restart functionality of flow does not seem to work as expected

### ISSUE THREAD

**Issue by blattms** [OPM/opm-simulators#720]
I just tested the restart functionality with model number 2 and it does not work.

I did a complete run with

```
RPTRST
  BASIC=2 FREQ=6 ALLPROPS  CONV=10  RVSAT RSSAT PBPD  NORST=1   /
--RPTRST
--  BASIC=5 FREQ=6 ALLPROPS  CONV=10  RVSAT RSSAT PBPD  NORST=1   /
--ALLPROPS --> fluid densities, viscosities , reciprocal formation volume factors and phase relative permeabilities
--NORST=1  --> output for visualization only 
```

and then tried to restart at simulation step 219. The adaptive time stepper tries to use 10.23 days (original run
31 days) and fails miserably after 10 times reducing the timestep (final size: 0.00156681) with:

```
WARNING: Failed to compute converged solution in 15 iterations.
Overall linear iterations used: -1
ESC[31;1m[/home/mblatt/DUNE-test/opm-core/opm/core/simulator/AdaptiveTimeStepping_impl.hpp:242]
Solver failed to converge after 10 restarts.ESC[0m
Program threw an exception:
[/home/mblatt/DUNE-test/opm-core/opm/core/simulator/AdaptiveTimeStepping_impl.hpp:242]
Solver failed to converge after 10 restarts.
```

I guess that should not happen...

**Comment by blattms** [comment]
Not sure whether this is the whole cause of this problem, but at least for this model the geology update (ScheduleEvents::GEO_MODIFIER) that happens before this step in  a full simulation is neglected for a run restarted after that event.

**Comment by blattms** [comment]
The same problem occurs if start before the modification of the Geology. (simulation step 108). I might check some easier test cases over the weekend.

**Comment by blattms** [comment]
I miserably fail to test this with easier test cases. How it works is totally opaque to me.

I tried the same modifications (RPTSTP BASIC=2) with SPE9, but always can the following error when trying to the restart simulation:
```
Program threw an exception: Restart file ./restart_test/SPE9_BASIC2.X0036 not found!
```

**Comment by blattms** [comment]
Changing keyword UNIFOUT to UNIFIN creates the *.X* files. That is a bit confusing. I interpreted the explanation in http://folk.uib.no/fciop/index_htm_files/ResSimNotes.pdf the other way around.

**Comment by blattms** [comment]
For SPE9 restarting works and creates the same plots.

**Comment by joakim-hove** [comment]
It is a bit unclear to me what the status really is here; but I must admit that my general confidence with the restart functionality is a bit like: "We have got it to work at least once on a SPE case". There will certainly be more issues here.

**Comment by blattms** [comment]
This restart functionality is fas to complicated for me. For norne I have
```
SOLUTION

RPTRST
BASIC=2 KRO KRW KRG /
```
and still get an exception:
```sh
Program threw an exception: Restart file ./norne-restart-test/NORNE_ATW2013.UNRST does not contain data for report step 150!
```

Furthermore comments in the eclipse file like
```
---------------------------------------------------------------------------------
-- equilibrium data: do not include this file in case of RESTART

INCLUDE
'./INCLUDE/PETRO/E3.prop' /
```
really make me uneasy. Reads like I need to modify files even more for restarting.

**Comment by alfbr** [comment]
Please mail me the DATA files you are using on model 2, and I can check at my end that they are supposed to work.

**Comment by atgeirr** [comment]
I am pretty sure that the restart functionality was broken after OPM/opm-output#30. There may be a fix in OPM/opm-output#44, at least I no longer experience segfaults in the output although it may not be sufficient to fix any of the problems you are having.

**Comment by andlaus** [comment]
this should have been fixed. reopen the issue if not.
